### PR TITLE
Move is app subscribed for way points

### PR DESCRIPTION
**Pull request by VVeremjova**
This pull is moved from PASA to GENIVI repo. PR on PASA - #344
Previously this PR was created on GENIVI  - #597

Related to [APPLINK-24305](https://adc.luxoft.com/jira/browse/APPLINK-24305)

### CODE REVIEW

**Comment by VVeremjova** [comment]
Please review @OHerasym, @okozlovlux, @LevchenkoS, @VProdanov, @dtrunov, @nk0leg

**Comment by LevchenkoS** [comment]
@VVeremjova, reviewed.

**Comment by dtrunov** [comment]
Reviewed

File: src/components/application_manager/include/application_manager/application_impl.h [hunk 1]
```diff
@@ -326,6 +326,9 @@ class ApplicationImpl : public virtual InitialApplicationDataImpl,
    */
   uint32_t GetAvailableDiskSpace() OVERRIDE;
 
+  virtual bool IsAppSubscribedForWayPoints(
+      const uint32_t app_id) const OVERRIDE;
```
Comment (VProdanov) [inline src/components/application_manager/include/application_manager/application_impl.h]: @VVeremjova virtual is redundant

File: src/components/application_manager/test/include/application_manager/mock_resumption_data.h [hunk 2]
```diff
@@ -37,6 +37,7 @@
 #include "application_manager/application.h"
 #include "application_manager/mock_application_manager_settings.h"
 #include "application_manager/mock_application_manager.h"
+#include "application_manager/mock_application_manager_settings.h"
```
Comment (VProdanov) [inline src/components/application_manager/test/include/application_manager/mock_resumption_data.h]: @VVeremjova dublicate include

File: src/components/application_manager/test/include/application_manager/test_resumption_data_db.h [hunk 3]
```diff
@@ -37,6 +37,7 @@
 #include "application_manager/resumption/resumption_data_db.h"
 #include "application_manager/mock_application_manager_settings.h"
 #include "application_manager/mock_application_manager.h"
+#include "application_manager/mock_application_manager_settings.h"
```
Comment (VProdanov) [inline src/components/application_manager/test/include/application_manager/test_resumption_data_db.h]: @VVeremjova  included twice

**Comment by VProdanov** [comment]
@VVeremjova Reviewed

File: src/components/application_manager/test/include/application_manager/mock_resumption_data.h (hunk 2 above)
Comment (VVeremjova) [inline src/components/application_manager/test/include/application_manager/mock_resumption_data.h]: @VProdanov See 8d57369

File: src/components/application_manager/include/application_manager/application_impl.h (hunk 1 above)
Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/application_impl.h]: @VProdanov See 8d57369

File: src/components/application_manager/test/include/application_manager/test_resumption_data_db.h (hunk 3 above)
Comment (VVeremjova) [inline src/components/application_manager/test/include/application_manager/test_resumption_data_db.h]: @VProdanov See 8d57369

**Comment by OHerasym** [comment]
Reviewed

**Comment by okozlovlux** [comment]
@VVeremjova, reviewed

**Comment by nk0leg** [comment]
@VVeremjova, reviewed.

**Pull request merged**